src/vsr.zig

```diff
@@ -999,14 +1004,29 @@ pub fn quorums(replica_count: u8) struct {
     // The insight is that the replication phase is by far more common than the view change.
     // This trade-off allows us to optimize for the common case.
     // See the comments in `constants.zig` for further explanation.
+    assert(quorum_view_change >= 2 or quorum_view_change == replica_count);
     assert(quorum_view_change >= majority);
+    assert(quorum_view_change + quorum_replication > replica_count);
 
     return .{
         .replication = quorum_replication,
         .view_change = quorum_view_change,
     };
 }
 
+test "quorums" {
+    if (constants.quorum_replication_max != 3) return;
+
+    const expect_replication = [_]u8{ 1, 2, 2, 2, 3, 3 };
+    const expect_view_change = [_]u8{ 1, 2, 2, 3, 3, 4 };
```

Comment:
```suggestion
    const expect_replication = [_]u8{ 1, 2, 2, 2, 3, 3, 3, 3 };
    const expect_view_change = [_]u8{ 1, 2, 2, 3, 3, 4, 5, 6 };
```

so that we actually hit the `quorum_replication_max` branches.

This also makes me think that we don't need `quorum_view_change = std.math.max`, and that we can rather assert that its >= majority

Reply:
Ah, or would it not work due to our replicas_max=6?

Reply:
`quorums()` doesn't assert `replicas_max` so it still works.